client/orm/qb/delete.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 beego. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qb
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/valyala/bytebufferpool"
+
+	"github.com/beego/beego/v2/client/orm"
+	"github.com/beego/beego/v2/client/orm/internal/models"
+)
+
+var _ QueryBuilder = &Deleter[any]{}
+
+// Deleter builds DELETE query
+type Deleter[T any] struct {
+	builder
+	table interface{}
+	db    orm.Ormer
+	where []Predicate
+}
+
+// NewDeleter 开始构建一个 DELETE 查询
+func NewDeleter[T any](db orm.Ormer) *Deleter[T] {
+	return &Deleter[T]{
+		db: db,
+		builder: builder{
+			buffer: bytebufferpool.Get(),
+		},
+	}
+}
+
+func (d *Deleter[T]) Build() (*Query, error) {
+	defer bytebufferpool.Put(d.buffer)
+	d.writeString("DELETE FROM ")
+	var err error
+	if d.table == nil {
+		d.table = new(T)
+	}
+	registry := models.DefaultModelCache
+	d.model, _ = registry.GetByMd(d.table)
+	if d.model == nil {
+		err = registry.Register("", true, d.table)
+		if err != nil {
+			return nil, err
+		}
+		d.model, _ = registry.GetByMd(d.table)
+	}
```

Comment:
can you provide a new function which will register the model if model not found?
```go
func (m model) GetOrRegister() {
   // ...
}
```
I remember that Beego already has a similar function, you can check it.